featuretools/selection/selection.py

```diff
@@ -18,3 +21,132 @@ def remove_low_information_features(feature_matrix, features=None):
+                        If null, defaults to checking all columns.
+            features_to_keep (list[str], optional): List of colum names to keep even if correlated to another column.
+                        If null, all columns will be candidates for removal.
+
+        Returns:
+            pd.DataFrame, list[:class:`.FeatureBase`]:
+                The feature matrix and the list of generated feature definitions. Matches dfs output.
+                If no feature list is provided as input, the feature list will not be returned.
+                For consistent results, do not change the order of features outputted by dfs.
+    """
+    if pct_corr_threshold < 0 or pct_corr_threshold > 1:
+        raise ValueError("pct_corr_threshold must be a float between 0 and 1, inclusive.")
+
+    if features_to_check is None:
+        features_to_check = feature_matrix.columns
+    else:
+        for f_name in features_to_check:
+            assert f_name in feature_matrix.columns, "feature named {} is not in feature matrix".format(f_name)
+
+    if features_to_keep is None:
+        features_to_keep = []
+
+    boolean = ['bool']
+    numeric_and_boolean_dtypes = vtypes.PandasTypes._pandas_numerics + boolean
+
+    fm_to_check = (feature_matrix[features_to_check]).select_dtypes(include=numeric_and_boolean_dtypes)
+
+    dropped = set()
+    columns_to_check = fm_to_check.columns
+    # When two features are found to be highly correlated, we drop the more complex feature
+    # Columns produced later in dfs are more complex
+    for i in range(len(columns_to_check) - 1, 0, -1):
+        more_complex_name = columns_to_check[i]
+        more_complex_col = fm_to_check[more_complex_name]
+
+        for j in range(i - 1, -1, -1):
+            less_complex_name = columns_to_check[j]
+            less_complex_col = fm_to_check[less_complex_name]
+
+            if abs(more_complex_col.corr(less_complex_col)) >= pct_corr_threshold:
+                dropped.add(more_complex_name)
+                break
+
+    keep = [f_name for f_name in feature_matrix.columns if (f_name in features_to_keep or
+                                                            f_name not in dropped)]
+    return _apply_feature_selection(keep, feature_matrix, features)
+
+
+def _apply_feature_selection(keep, feature_matrix, features=None):
+    new_matrix = feature_matrix[keep]
+    new_feature_names = set(new_matrix.columns)
+
+    if features is not None:
+        sliced_features = []
+        for f in features:
+            if (f.number_output_features == 1 or
+                    all([slice_name in new_feature_names for slice_name in f.get_feature_names()])):
+                sliced_features.append(f)
+            else:
+                sliced_features.extend([f[i] for i in range(f.number_output_features)])
+        new_features = [f for f in sliced_features if f.get_name() in new_feature_names]
```

Comment:
if `new_feature_names` is a set of column names, wouldn't `f.get_name()` fail for a multi-output feature?